proto/example.go

```diff
@@ -3,27 +3,43 @@
 package main
 
 import (
-	"encoding/json"
 	"log"
+	"os"
 
 	oci "./go/"
+	"github.com/golang/protobuf/jsonpb"
 	"github.com/golang/protobuf/proto"
+	google_protobuf "google/protobuf"
 )
 
 func main() {
-	s := oci.LinuxSpec{
+	s := &oci.LinuxSpec{
 		Spec: &oci.Spec{
-			Platform: &oci.Platform{Os: proto.String("linux"), Arch: proto.String("x86_64")},
+			Platform: &oci.Platform{Os: "linux", Arch: "x86_64"},
 			Process: &oci.Process{
-				Cwd: proto.String("/"),
+				Cwd: "/",
 				Env: []string{"TERM=linux"},
 			},
 		},
 	}
 
-	buf, err := json.MarshalIndent(s, "", "  ")
+	user := &oci.LinuxUser{Uid: 1, Gid: 1, AdditionalGids: []int32{5, 6}}
+	userBytes, err := proto.Marshal(user)
+	if err != nil {
+		log.Fatal(err)
+	}
+	s.Spec.Process.User = &google_protobuf.Any{
+		TypeUrl: "oci.LinuxUser",
+		Value:   userBytes,
+	}
+
+	marshaler := jsonpb.Marshaler{
+		EnumsAsString: true,
```

Comment:
On Sat, Oct 24, 2015 at 10:40:33AM -0700, Peter Edge wrote:
> > +		EnumsAsString: true,
> 
> golang/protobuf just removed this
> https://github.com/golang/protobuf/commits/master so this won't
> compile (has been the bane of my docker image existence for a few
> days :) )

Heh, two days between making it the (golang/protobuf@9ebc6c4ed9) and
removing it completely (golang/protobuf@8a5d8e8b) is frankly
rediculous :p.  No wonder the Go world is a fan of vendoring, but how
folks manage this sort of thing without a library release process is a
mystery to me ;).  Anyhow, fixed in 8214db4, and thanks for pointing
it out :).